lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -671,7 +671,7 @@ private static Optional<Collection<Object>> extractDiscreteValues(int domainComp
         return Optional.of(valueSet.getDiscreteSet());
     }
 
-    private FilterPredicate convertToParquetFilter(DateTimeZone timeZone)
+    public FilterPredicate convertToParquetFilter(DateTimeZone timeZone)
```

Comment:
Undo

This method is not good to expose, since it returns nullable.

use `toParquetFilter` defined above which delegates here and wraps with optional